kernel/src/scan/mod.rs

```diff
@@ -388,13 +388,35 @@ impl std::fmt::Debug for Scan {
 }
 
 impl Scan {
-    /// Get a shared reference to the [`Schema`] of the scan.
+    /// The table's root URL. This is the location where data files with relative paths will
+    /// reside.
```

Comment:
```suggestion
    /// The table's root URL. Any relative paths returned from `scan_data` (or in a callback from [`state::visit_scan_files`]) must be resolved against this root to get the actual path to the file.
```

Reply:
ah nice, thanks!

Reply:
just realized this should probably be `visit_scan_metadata` or something (not `visit_scan_files`) - will do a separate rename PR